lambda/custom/helperFunctions.js

```diff
@@ -887,6 +890,105 @@ const postDirectMessage = async (message, roomid, headers) =>
+
+		let privateChannelsResponse = await axios.get(apiEndpoints.grouplisturl, {
+			headers
+		}).then((res) => res.data)
+
+		// adding public channels to the array
+		let channels = publicChannelsResponse.channels.map(channel => {
+			return {
+				name: channel.name,
+				id: channel._id,
+				type: channel.t}
+			})
+
+		// adding private channels to the array	
+		channels = channels.concat(privateChannelsResponse.groups.map(channel => {
+			return {
+				name: channel.name,
+				id: channel._id,
+				type: channel.t
+			}
+		}))
+
+		// this part returns only one channel which best matches the input string
+		if (single){
+			let channelNames = channels.map(channel => channel.name)
+			let channel = stringSimilar.findBestMatch(channelName, channelNames).bestMatch.target
+			return channels.find((elem) => elem.name == channel)
+		}
+
+		// using string similarity module to filter channels and groups which are similar to the input string
+		// here compareTwoStrings returns a rating which indicates how closely the strings match
+		// 0.3 is a rating being used here, which can be adjusted according to our needs
+		let similarChannels = channels.filter((channel) => stringSimilar.compareTwoStrings(channelName, channel.name) > settings.similarity_index )    
+		return similarChannels
+	}catch (err) {
+		console.log(err)
+	}
+}
+
+/* 
+this function takes a string an an input and returns an array of usernames 
+which the user is in contact with and is similar to the input string
+*/
+const resolveUsername = async (username, headers, single = false) => {
+	try{
+		let subscriptions = await axios.get(apiEndpoints.getsubscriptionsurl, {
+			headers
+		})
+		.then(res => res.data.update)
+		// the getsubscriptionsurl returns all subscriptions including private and public channels
+		// since we only need to resolve usernames we filter only direct message subscriptions
+		.then(subscriptions => {
+			return subscriptions.filter(subscription => {
+				return subscription.t == 'd' 
+			})
+		})
+		.then(subscriptions => {
+			return subscriptions.map(subscription => {
+				return {
+					name: subscription.name,
+					id: subscription._id,
+					type: subscription.t
+				}
+			})
+		})
+
+		// this part returns only one username which best matches the input string
+		if (single){
+			let usernames = subscriptions.map(subscription => subscription.name)
+			let user = stringSimilar.findBestMatch(username, usernames).bestMatch.target
+			return subscriptions.find((elem) => elem.name == user)
+		}
+
+		// using string similarity module to filter usernames matching the input string
+		let similarUsernames = subscriptions.filter((subscription) => stringSimilar.compareTwoStrings(username, subscription.name) > settings.similarity_index )
+		return similarUsernames
+
+	}catch(err){
+		console.log(err)
+	}
+}
+
+// this functions logs the data to an external site
+const customLog = async (data) => {
+	try{
+		axios.post('http://my-logs.glitch.me/', (data))
```

Comment:
```suggestion
		axios.post('http://my-logs.glitch.me/', (data))
		
    module.exports = {
		serverurl: "https://bots.rocket.chat/",
		oauthServiceName: "alexaskill",
		dynamoDBTableName: "alexa",
	};
```

I don't think we should do this, as not everyone will be using the same URL as we are using. Make sure it is easily configurable by anyone who is forking our repo and not hardcoded.

Reply:
Moved all these to .env variable which is not uploaded to github.
Updated README with setup instructions.